_posts/2024-08-28-Data-Prepper-2.9.0-is-ready-for-download.md

```diff
@@ -0,0 +1,140 @@
+
+Many pipeline authors may these processors and then remove the source data in a second processor.
+This is a good approach when you don't need to store the original string in your sink.
+But it doesn't always make the memory used by the string available for garbage collection when you need it.
+The reason for this is that Data Prepper pipelines operate on batches of data.
+As these batches of data move through the pipeline, the pipeline will expand the memory usage in one processor and then attempt to reduce it in the next.
+Because the memory expansion happens in batches, Data Prepper may expand many thousands of events before starting to remove the source data.
+
+See the following example pipeline, which runs `grok` and then `delete_entries`.
+With a configured `batch_size` of 100,000, Data Prepper will expand 100,000 events before deleting the messages.
+
+```
+my-pipeline:
+  buffer:
+    bounded_blocking:
+      batch_size: 100000
+  processor:
+    - grok:
+        match:
+          message: ["..."]
+    - delete_entries:
+        with_keys: ["message"]
+```
+
+To help with this memory usage issue, Data Prepper now provides a `delete_source` flag on some of these processors, including `grok` and `parse_json`.
+
+Returning to the preceding example, you could both simplify the pipeline and reduce the amount of memory used in between processors:
+
+```
+my-pipeline:
+  buffer:
+    bounded_blocking:
+      batch_size: 100000
+  processor:
+    - grok:
+        match:
+          message: ["..."]
+        delete_source: true
+```
+
+If you observe this pattern of the source being deleted in a separate processor, configure your pipeline to use `delete_source` in order to improve your overall memory usage.
+
+
+## Getting started
+
+* To download Data Prepper, visit the [OpenSearch downloads](https://opensearch.org/downloads.html) page.
+* For instructions on how to get started with Data Prepper, see [Getting started with Data Prepper](https://opensearch.org/docs/latest/data-prepper/getting-started/).
+* To learn more about the work in progress for Data Prepper 2.10 and other releases, see the [Data Prepper roadmap](https://github.com/opensearch-project/data-prepper/projects/1).
+
+## Thanks to our contributors!
+
+The following people contributed to this release. Thank you!
+
+* [chenqi0805](https://github.com/chenqi0805) - Qi Chen
+* [danhli](https://github.com/danhli) - Daniel Li
+* [dinujoh](https://github.com/dinujoh) - Dinu John
+* [dlvenable](https://github.com/dlvenable) - David Venable
+* [graytaylor0](https://github.com/graytaylor0) - Taylor Gray
+* [ivan-tse](https://github.com/ivan-tse) - Ivan Tse
+* [jayeshjeh](https://github.com/jayeshjeh) - Jayesh Parmar
+* [joelmarty](https://github.com/joelmarty) - Joël Marty
+* [kkondaka](https://github.com/kkondaka) - Krishna Kondaka
+* [mishavay-aws](https://github.com/mishavay-aws) - None
```

Comment:
I probably wouldn't include "None" here and in the last line.

Reply:
This is the output from the GitHub API, but I'll change it.